frontend/src/store/modules/visitors/actions.js

```diff
@@ -9,11 +24,31 @@ export default {
         context.commit(SET_ACTIVE_BUTTON, button);
     },
     [CHANGE_FETCHED_BUTTON_STATE]: (context, data) => {
-
         if (data.value) {
             context.commit(SET_BUTTON_FETCHING, data.button);
         } else {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [GET_BUTTON_DATA]: (context, data) => {
+        if (data.value) {
+            context.commit(SET_BUTTON_FETCHING, data.button);
+            context.commit(GET_SELECTED_PERIOD)
+                .then(response => {
+                    periodService.getTimeByPeriod(response.data);
+                })
+                .then(response => {
+                    factoryVisitorService.create(data.button)
+                        .fetchButtonValue(response.startDate, response.endDate);
+                })
+                .then(response => {
+                    context.commit(SET_BUTTON_DATA, data.button, response.data);
+                })
+                .catch(err => {
+                    context.commit(RESET_BUTTON_FETCHING, data.button);
+                    throw err;
+                });
+            context.commit(RESET_BUTTON_FETCHING, data.button);
```

Comment:
reset fetching after you got successful lresult. You can use .finally() of the promise